### PR TITLE
docs(guides): update against webpack 5

### PR DESCRIPTION
**Pull request by chenxsan**
closes https://github.com/webpack/webpack.js.org/issues/4090 fixes https://github.com/webpack/webpack.js.org/issues/4098

`Guide` section needs to be updated against webpack 5, I'll do it in this pull request.

PREVIEW URL: https://webpack-js-org-git-feature-update-guide-against-webpack5.webpack-docs.vercel.app/guides/getting-started/

- [x] update getting-started.md (code demo https://github.com/chenxsan/webpack-demo)
- [x] update asset-management.md (code demo https://github.com/chenxsan/webpack-demo/pull/2)
- [ ] update output-management.md (code demo https://github.com/chenxsan/webpack-demo/pull/3)

### CODE REVIEW

**Review by evilebottnawi — APPROVED** [review]
Thanks

File: src/content/guides/asset-management.md [hunk 1]
```diff
@@ -90,7 +87,7 @@ __webpack.config.js__
 
 Module loaders can be chained. Each loader in the chain applies transformations to the processed resource. A chain is executed in reverse order. The first loader passes its result (resource with applied transformations) to the next one, and so forth. Finally, webpack expects JavaScript to be returned by the last loader in the chain. 
 
-The above order of loaders should be maintained: 'style-loader' comes first and followed by 'css-loader'. If this convention is not followed, webpack is likely to throw errors.
+The above order of loaders should be maintained: `'style-loader'` comes first and followed by `'css-loader'`. If this convention is not followed, webpack is likely to throw errors.
```
Comment (evilebottnawi) [inline src/content/guides/asset-management.md:90]: I think we should write the plugin to lint this, ideally we should always use "`" and provide link

Comment (chenxsan) [reply src/content/guides/asset-management.md:90]: Single quote is preferred for code https://webpack.js.org/contribute/writers-guide/#quotation but there's no rule for text so far as I know.

**Comment by chenxsan** [comment]
I found It's very difficult to maintain those diff blocks, any suggestion?

**Comment by chenxsan** [comment]
`webpack serve --open` won't work at the moment as there's [a bug](https://github.com/webpack/webpack.js.org/pull/4091#discussion_r512003420) in webpack-cli:

```
$ webpack serve --open
error: option '--open <value>' argument missing
```
I'll continue this PR after it's fixed.

**Comment by snitin315** [comment]
@chenxsan Can you open an issue in `webpack-cli` to keep track of this ?

File: src/content/guides/asset-management.md [hunk 2]
```diff
@@ -336,37 +297,47 @@ With the loader configured and fonts in place, you can incorporate them via an `
...
-+   font-style: normal;
-+ }
-
-  .hello {
-    color: red;
-+   font-family: 'MyFont';
-    background: url('./icon.png');
-  }
++@font-face {
++  font-family: 'GoogleFont';
```
Comment (montogeek) [inline src/content/guides/asset-management.md:301]: Why this change? People may get confused and will start downloading GoogleFonts instead of using their CDN.

I think this example is targeted for use cases when your font is not available anywhere and you have to ship ip with your project.

File: src/content/guides/asset-management.md [hunk 3]
```diff
@@ -380,50 +351,20 @@ npm install --save-dev csv-loader xml-loader
...
-+       {
-+         test: /\.xml$/,
-+         use: [
-+           'xml-loader',
-+         ],
-+       },
-      ],
-    },
-  };
+         test: /\.(woff|woff2|eot|ttf|otf)$/i,
```
Comment (montogeek) [inline src/content/guides/asset-management.md:354]: I think it is better to keep the entire diff, in this case you are replacing the entire content, so it is not longer a diff.

File: src/content/guides/asset-management.md [hunk 4]
```diff
@@ -573,57 +505,64 @@ And configure them in your webpack configuration:
...
-        type: 'json',
-        parser: {
-          parse: json5.parse
-        }
-      }
-    ]
-  }
-};
+```diff
+ const path = require('path');
```
Comment (montogeek) [inline src/content/guides/asset-management.md:509]: This is inverted, `path` is the new line added.

File: src/content/guides/asset-management.md [hunk 5]
```diff
@@ -573,57 +505,64 @@ And configure them in your webpack configuration:
...
-console.log(toml.title); // output `TOML Example`
-console.log(toml.owner.name); // output `Tom Preston-Werner`
-
-console.log(yaml.title); // output `YAML Example`
-console.log(yaml.owner.name); // output `Tom Preston-Werner`
-
-console.log(json.title); // output `JSON5 Example`
-console.log(json.owner.name); // output `Tom Preston-Werner`
+```diff
+ import Icon from './icon.png';
```
Comment (montogeek) [inline src/content/guides/asset-management.md:547]: Also inverted, Icon, Data and Notes are new lines.

File: src/content/guides/development.md [hunk 6]
```diff
@@ -26,28 +27,13 @@ Let's start by setting [`mode` to `'development'`](/configuration/mode/#mode-dev
 __webpack.config.js__
 
 ``` diff
-  const path = require('path');
-  const HtmlWebpackPlugin = require('html-webpack-plugin');
```
Comment (montogeek) [inline src/content/guides/development.md:30]: Why removing it?

File: src/content/guides/development.md [hunk 7]
```diff
@@ -143,59 +119,28 @@ Let's add an npm script that will start webpack's Watch Mode:
...
-      "xml-loader": "^1.2.1"
-    }
-  }
+   "private": true,
+   "scripts": {
+     "test": "echo \"Error: no test specified\" && exit 1",
++    "watch": "webpack --watch",
+     "build": "webpack"
+   },
+   "repository": {
```
Comment (montogeek) [inline src/content/guides/development.md:128]: Better to keep the entire diff.

File: src/content/guides/getting-started.md [hunk 8]
```diff
@@ -40,7 +40,20 @@ npm init -y
 npm install webpack webpack-cli --save-dev
 ```
 
-T> Throughout the Guides we will use `diff` blocks to show you what changes we're making to directories, files, and code.
+Throughout the Guides we will use __`diff` blocks__ to show you what changes we're making to directories, files, and code. For instance:
```
Comment (montogeek) [inline src/content/guides/getting-started.md:43]: Entire point of a diff it so show the difference between 2 chucks of code, removing the deleted changes doesn't really follow its purpose.
Please keep the entire diffs in all examples.

Comment (chenxsan) [reply src/content/guides/getting-started.md:43]: Do you have any idea how to keep the entire diffs? Currently I'm using `git diff A..B > diff.txt`, which results in all those changes.

**Comment by chenxsan** [comment]
I'm dropping this PR in favor of other smaller ones as this one will be a larger one making it difficult to review.
